packages/sparrowql/lib/utils.js

```diff
@@ -63,7 +63,7 @@ function getNameRelative(prefix, name, isPath) {
             .split('.')
             .slice(1)
             .join('.');
-    } else if (name.startsWith(prefix)) {
+    } else if (!isRelation && name.startsWith(prefix)) {
```

Comment:
I think `name.startsWith(`${prefix}.`)` would be better. `getNameRelative` is only used with Mongo paths, and adding a new parameter adds unnecessary complexity.

Reply:
Are you sure this is a good solution? AFAIK collection names can include dots and this bug would still be present in some edge cases.

Reply:
Well, yes and no. Technically yes, a collection name can include dots, but SparrowQL doesn't work with such names anyway, due to the way how the rest is implemented. I'd go with that here and implement the handling of dots in names separately.

Reply:
I'm not sure I like your suggestion though as not only it makes this PR incomplete because there technically are still edge cases left and leaves these cases still to be fixed and requires introducing more logic. 

What I was debating though was exporting part of this logic that only applies to the collection names to a separate functions and reuse it in `getNameRelative` and `$lookup` and `$unwind` parts of the code but I was unsure on which logic should be exported.

Reply:
I didn't want to get into this now, but OK. So using dots in collection names will break everything, because:
* We're using these names in `$group` stage for joined collections, and `{$group: {'a.b': ...}}` is invalid.
* Similarly, `$project` would create unnecessarily nested paths. While It should work most of the time, it may cause problems with arrays.

These two are from the top of my head, but I think there'll be more. That's why I think we should skip it now and get back to proper support later. Probably replacing dots with something else here and there should be enough.

Reply:
I just tried your suggested solution and unfortunately it breaks almost all the tests.

What I thought would be the best solution for this problem is to extract some of the logic to a separate function because it makes more sense - field paths and collection names should be treated differently and should not share the same logic. But I was unsure which logic is required by what.

Reply:
After discussion we came up with a new solution by adding a dot if it was not present. Changed.